src/components/Dashboard/Analytics/AnalyticsBarChart.jsx

```diff
@@ -0,0 +1,32 @@
+"use client";
+
+import React, { useState, useEffect } from "react";
+import { Bar } from "react-chartjs-2";
+import { VerticalBarChartContainer } from "./AnalyticsBarChartElements";
+import { Chart as ChartJS, CategoryScale, LinearScale, BarElement, Title, Tooltip, Legend } from "chart.js";
+import { displayMonths, barChartOptions as options, smallBarChartDatasets as datasets } from "./AnalyticsUtils";
+
+ChartJS.register(CategoryScale, LinearScale, BarElement, Title, Tooltip, Legend);
+
+export default function BarChart() {
```

Comment:
ok, rule of thumb, if you copy past something more then once and it is a small block of code, we can "forgive" but if you copy past something more then once , it says that there have to be a better way, and you have to find it.
AnalyticsDoughnutChart , AnalyticsBarChart , AnalyticsMainBarChart  all these components share very similar code, combine them to be one by using props , through the props you can send different styles / datasets/ options/ chart and so on.

Reply:
Noted. I will do that.